tests_e2e/tests/agent_update/rsm_update.py

```diff
@@ -241,6 +247,17 @@ def _verify_agent_reported_update_status(self, version: str):
         self._ssh_client.run_command(f"agent_update-verify_agent_reported_update_status.py --version {version}", use_sudo=True)
         log.info("Successfully Agent reported update status for version {0}".format(version))
 
+    def _retrieve_installed_agent_version(self):
+        """
+        Retrieve the installed agent version
+        """
+        log.info("Retrieving installed agent version")
+        stdout: str = self._ssh_client.run_command("waagent-version", use_sudo=True)
+        log.info("Retrieved installed agent version \n {0}".format(stdout))
+        match = re.search(r'.*Goal state agent: (\S*)', stdout)
+        if match:
```

Comment:
no match should be an error instead of defaulting to 9.9.9.9, shouldn't it

Reply:
Even we raise an error, we still check ignore rules.  Need default value either case but that won't change behavior and makes if condition false, so rule won't apply.

But I'll raise error

Reply:
or at least a warning, rather than silently default to 9.9.9.9